crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -534,6 +543,157 @@ impl kitsune2_api::KitsuneHandler for HolochainP2pActor {
             }
         })
     }
+
+    fn preflight_gather_outgoing(&self, _peer_url: Url) -> K2Result<bytes::Bytes> {
+        Ok(self.preflight.lock().unwrap().clone())
+    }
+
+    fn preflight_validate_incoming(&self, _peer_url: Url, data: bytes::Bytes) -> K2Result<()> {
+        // decode the preflight that the remote sent us
+        let rem = crate::wire::WirePreflightMessage::decode(&data)
+            .map_err(|err| K2Error::other_src("Invalid remote preflight", err))?;
+
+        // if the compats don't match, reject the connection
+        if rem.compat != self.compat {
+            return Err(K2Error::other(format!(
+                "Invalid remote preflight, wanted {:?}, got: {:?}",
+                self.compat, rem.compat
+            )));
+        }
+
+        let mut agents = Vec::new();
```

Comment:
Allocate?

Reply:
Oh I guess we might not want to, there much be a bunch of invalid agents.

Reply:
Below we abort if there is an error, so in happy path the lens should equal. We can allocate here.

```suggestion
        let mut agents = Vec::with_capacity(rem.agents.len());
```